### PR TITLE
[Layout foundations] Clean up prop descriptions and add default values in description

### LINKED ISSUE #7742: [Layout foundations] Clean up layout primitives to document default props and examples with JSDoc
Columns props as example:

Image

### PR DESCRIPTION
**Pull request by laurkim**
### WHY are these changes introduced?

Resolves #7742.

### WHAT is this pull request doing?

Cleans up prop descriptions and adds missing default values with JSDoc.
Updates the props in style guide to align with `get-props` script.

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [x] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris-react/src/components/Bleed/Bleed.tsx [hunk 1]
```diff
@@ -6,21 +6,21 @@ import {sanitizeCustomProperties} from '../../utilities/css';
 import styles from './Bleed.scss';
 
 export interface BleedProps {
-  /** Elements to display inside tile */
+  /** Elements to display inside bleed */
```
Comment (kyledurand) [inline polaris-react/src/components/Bleed/Bleed.tsx]: I still don't love that we have to document children

**Review by kyledurand — APPROVED** [review]

File: polaris-react/src/components/Bleed/Bleed.tsx (hunk 1 above)
Comment (laurkim) [reply polaris-react/src/components/Bleed/Bleed.tsx]: We can remove descriptions for children for all the layout primitives. I don't know if the description necessarily provides much value 😅

Comment (aveline) [reply polaris-react/src/components/Bleed/Bleed.tsx]: I agree I don't think it provides much value and we should remove it from all our components.

**Review by aveline — APPROVED** [review]
🎉

File: polaris-react/src/components/Box/Box.tsx [hunk 2]
```diff
@@ -168,12 +171,10 @@ export interface BoxProps {
    * paddingInlineEnd={{xs: '2', sm: '3', md: '4', lg: '5', xl: '6'}}
    */
   paddingInlineEnd?: Spacing;
-  /** Shadow */
+  /** Shadow color on box */
```
Comment (sarahill) [inline polaris-react/src/components/Box/Box.tsx]:
```suggestion
  /** Shadow on box */
```
Should this just be `Shadow on box`?

Comment (laurkim) [reply polaris-react/src/components/Box/Box.tsx]: Ah yup that was a typo on my part.

**Review by sarahill — APPROVED** [review]
LGTM! ✅ 
Left a minor comment but not a blocker

**Comment by laurkim** [comment]
I'll hold off on merging this so that it doesn't cause conflicts on the current Inline and Box prs.
I'll fix merge conflicts on this PR afterwards @aveline @kyledurand

**Pull request merged**